asserts/gpgkeypairmgr.go

```diff
@@ -23,15 +23,51 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	"os"
 	"os/exec"
+	"path/filepath"
+	"strconv"
 	"strings"
+
+	"github.com/snapcore/snapd/osutil"
 )
 
+func ensureGPGHomeDirectory(homedir string) (string, error) {
+	real, err := osutil.RealUser()
+	if err != nil {
+		return "", err
+	}
+
+	uid, err := strconv.Atoi(real.Uid)
+	if err != nil {
+		return "", err
+	}
+
+	gid, err := strconv.Atoi(real.Gid)
+	if err != nil {
+		return "", err
+	}
+
+	if homedir == "" {
+		homedir = os.Getenv("SNAP_GNUPGHOME")
+	}
+	if homedir == "" {
+		homedir = filepath.Join(real.HomeDir, ".snap")
```

Comment:
Can we please move this under ~/.snap/gnupg/, so we don't risk any surprising conflicts.